### PR TITLE
[BAU-61] feat: implement osano in forma 36 website

### PR DESCRIPTION
**Pull request by gui-santos**
# Purpose of PR

We would like to implement some basic analytics to Forma’s website just to check the number of visits
And because of that we will need to add some form of cookie consent management
This PR implements Osano to manage that, since Forma 36 internal team has access to it

Screenshot 2021-08-04 at 15 52 57

To open the preferences panel I added the "Cookie Preferences" link in the footer of the website

Screenshot 2021-08-04 at 15 52 17

## Dotenv

Since we need to use environment variables to set up Osano, I add this dependency to be able to develop it locally
(I still need to set it up in Netlify)

to develop and check Osano, a developer will need to set a `.env.development` file in `/forma-36-website` directory 
The file will look like this

```
OSANO_KEY={{This value can be found at Contentful’s Osano dashboard}}
```

Osano is set to show up the banner for now, so anyone can check how it is working. But before merging this PR I will set it to only show the banner once we implement some analytics

## Yarn 1.18.0.... nevermind

UPDATE: Apparently it works in the CI with yarn >= 1.21.0
so the problem is only in my computer hahaha

~I had to downgrade yarn because I was not being able to add `dotenv` or any other dependency to the website
Every time I tried to do it I was getting this error:~

Screenshot 2021-08-04 at 13 01 56

~After some research, I discovered this is an issue that some people are having when using yarn >= 1.19.0 with workspaces
https://github.com/yarnpkg/yarn/issues/7807#issuecomment-817161486 .~

~So they started recommending to use v1.18.0
Apparently they are still checking if yarn 2.0 has this issue or not~

## Osano in preview env

Apparently Osano will not work in preview, since Netlify has these dynamic domains, it will be hard to set this up
but you can test it locally 👌 

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/master/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

**Comment by suevalov** [comment]
@gui-santos do we need to have 147k lines of code with yarn v1.18.0?

File: .yarnrc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
+
+
+yarn-path ".yarn/releases/yarn-1.18.0.cjs"
```
Comment (suevalov) [inline .yarnrc:5]: can we just enforce using a concrete version without specifying copying it locally?

File: packages/forma-36-website/README.md [hunk 2]
```diff
@@ -16,6 +16,16 @@ For local development run `yarn` from the root of this repo to install all depen
 
 The documentation website is powered using [Gatsby v2](https://www.gatsbyjs.org). Use `yarn start` to launch the development server. This will create a live preview with hot module reloading at http://localhost:8000/.
 
+### Environment variables
+
+We don’t have many environment variables in Netlify currently,
+but to use them in development you will need to create a file named `.env.development` in `/forma-36-website` directory with the following values:
+
+```
+OSANO_KEY={{This value can be found at Contentful’s Osano dashboard}}
```
Comment (suevalov) [inline packages/forma-36-website/README.md:25]: Is it handled gracefully if to environment variables are not provided?

**Comment by gui-santos** [comment]
> @gui-santos do we need to have 147k lines of code with yarn v1.18.0?

Hahahaha I don't know
I just followed the instructions I found in yarn's github

if anyone want to help me to find a better solution I'm all ears 🙌

**Comment by suevalov** [comment]
@gui-santos just keep a concrete version in engines and that's it

File: packages/forma-36-website/README.md (hunk 2 above)
Comment (gui-santos) [reply packages/forma-36-website/README.md:25]: good point! I have to fix that probably
Let me work on that

**Comment by gui-santos** [comment]
> @gui-santos just keep a concrete version in engines and that's it

@suevalov what should I do now?

Screenshot 2021-08-04 at 16 57 11

**Comment by suevalov** [comment]
@gui-santos you need to update the environment variable in Netlify settings. Do you have access to it?

**Comment by gui-santos** [comment]
> @gui-santos you need to update the environment variable in Netlify settings. Do you have access to it?

@suevalov I already set up the variables in Netlify
The error is because I set in engines that we should use v1.18.0 but Netlify is trying to use v1.22.5
and I don't know where to change that

**Comment by suevalov** [comment]
@gui-santos there's a variable to specify yarn version on Netlify

**Comment by gui-santos** [comment]
> @gui-santos there's a variable to specify yarn version on Netlify

@suevalov the error I was talking is in CircleCI
In Netlify we have other errors

I added `YARN_VERSION` in Netlify and we still get errors
I will dm you the logs

**Comment by gui-santos** [comment]
@suevalov magic 🧙‍♂️ 

I remove my changes to then `engines` in package.json so the pipeline used yarn@1.22.4
and everything worked! 🙌 

I guess the problem to add dependencies is only happening locally for me then 🤷

File: packages/forma-36-website/src/components/Layout.css [hunk 3]
```diff
@@ -17,3 +17,212 @@ pre {
...
+  border: 2px solid var(--color-element-mid);
+  border-radius: var(--border-radius-medium);
+  line-height: var(--line-height-default);
+}
+
+.osano-cm-dialog .osano-cm-button.osano-cm-save:focus {
+  box-shadow: var(--glow-muted);
+}
+.osano-cm-dialog .osano-cm-button.osano-cm-save:hover {
+  background-color: var(--color-element-lightest);
```
Comment (suevalov) [inline packages/forma-36-website/src/components/Layout.css:78]: can you use new token values?

Comment (gui-santos) [reply packages/forma-36-website/src/components/Layout.css:78]: Good catch! Totally forgot about then!

We need to updated then in the experience-components too... I copied this css from there

**Review by suevalov — APPROVED** [review]

**Review by burakukula — APPROVED** [review]
🎉

**Pull request merged**